ouroboros-consensus-cardano/tools/db-analyser/Analysis.hs

```diff
@@ -434,6 +448,187 @@ traceLedgerProcessing
       mapM_ Debug.traceMarkerIO traces
       return $ newLedger
 
+{-------------------------------------------------------------------------------
+  Analysis: maintain a ledger state and time the five major ledger calculations
+  for each block:
+
+  0. Forecast.
+  1. Header tick.
+  2. Header application.
+  3. Block tick.
+  4. Block application.
+
+  We focus on these 5 operations because they are involved in:
+
+  - Chain syncing.
+  - Block forging.
+  - Block validation.
+
+-------------------------------------------------------------------------------}
+
+data BenchmarkLedgerOpsState blk = BenchmarkLedgerOpsState {
+       -- | RTS stats that correspond to either the point in time in which the
+       -- benchmarking process started, or the end of the previous block
+       -- application.
+       prevRtsStats    :: !GC.RTSStats
+       -- | Intermediate ledger state to which the next block will be applied.
+     , prevLedgerState :: !(ExtLedgerState blk)
+     }
+
+benchmarkLedgerOpsOutputPath :: String
+benchmarkLedgerOpsOutputPath = "ledger-ops-costs.csv" -- Do not forget to update the documentation if you update this.
+
+benchmarkLedgerOps ::
+  forall blk.
+     ( HasAnalysis blk
+     , LedgerSupportsProtocol blk
+     )
+  => Analysis blk
+benchmarkLedgerOps AnalysisEnv {db, registry, initLedger, cfg, limit} =
+    -- TODO: at the moment we do not offer a configuration option for changing
+    -- the location of the output file.
+    IO.withFile benchmarkLedgerOpsOutputPath IO.WriteMode $ \outFileHandle -> do
+
+      let separator = "\t"
+      IO.hPutStrLn outFileHandle $  DP.showHeaders separator
+                                 ++ "...era-specific stats"
+
+      rtsStats <- GC.getRTSStats
+      let st0 = BenchmarkLedgerOpsState rtsStats initLedger
+      chrono <- new
+
+      void $ processAll db registry GetBlock initLedger limit st0 (process outFileHandle separator chrono)
+  where
+    ccfg = topLevelConfigProtocol cfg
+    lcfg = topLevelConfigLedger   cfg
+
+    process ::
+         IO.Handle
+      -> String
+      -- ^ Separator for the data that is printed
+      -> Chronometer
+      -> BenchmarkLedgerOpsState blk
+      -> blk
+      -> IO (BenchmarkLedgerOpsState blk)
+    process outFileHandle separator chrono BenchmarkLedgerOpsState {prevRtsStats, prevLedgerState} blk = do
+        let slot = blockSlot      blk
+        (!tkLdgrView, tForecast) <- forecast            slot prevLedgerState
+        (!tkHdrSt,    tHdrTick)  <- tickTheHeaderState  slot prevLedgerState tkLdgrView
+        (!hdrSt',     tHdrApp)   <- applyTheHeader                           tkLdgrView tkHdrSt
+        (!tkLdgrSt,   tBlkTick)  <- tickTheLedgerState  slot prevLedgerState
+        (!ldgrSt',    tBlkApp)   <- applyTheBlock                                       tkLdgrSt
+
+        currentRtsStats <- GC.getRTSStats
+        let
+          slotDataPoint = DP.mkSlotDataPoint
```

Comment:
I think the risk of passing arguments here to `mkSlotDataPoint` in the wrong order drastically out-weighs the risk of forgetting to use `showData`. So I suggest eliminating `mkSlotDataPoint` and using record expression syntax here.

Reply:
Would this be overkill? 

```haskell
slotDataPoint = DP.mkSlotDataPoint
                            (realPointSlot rp) -- We're ok here, there's only one `Slot` type
                            (SlotGap   $ slot `slotCount` getTipSlot prevLedgerState)
                            (TotalTime $ GC.elapsed_ns         currentRtsStats `nsDiff` GC.elapsed_ns prevRtsStats)

newtype SlotGap = ...
```

Reply:
I think so, yes.